src/com/amazon/corretto/crypto/provider/keygeneratorspi/SecretKeyGenerator.java

```diff
@@ -0,0 +1,55 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider.keygeneratorspi;
+
+import javax.crypto.KeyGeneratorSpi;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SecretKeyGenerator extends KeyGeneratorSpi {
```

Comment:
What's Spi?

Reply:
Service Provider Interface: in JCA, classes used by client code, such as `KeyGenerator`, are called engine classes. [Engine classes delegate the operation to SPI classes that are implemented by a provider](https://docs.oracle.com/en/java/javase/11/security/howtoimplaprovider.html#GUID-B35E3749-0A12-42C3-BA0B-C444D7E140BB).